Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/eventregistration/ui/createevent/CreateEventTestViewModel.kt

```diff
@@ -0,0 +1,82 @@
+package de.rki.coronawarnapp.test.eventregistration.ui.createevent
+
+import androidx.lifecycle.MutableLiveData
+import com.google.protobuf.ByteString
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.eventregistration.events.HostedEvent
+import de.rki.coronawarnapp.eventregistration.events.toHostedEvent
+import de.rki.coronawarnapp.eventregistration.storage.repo.HostedEventRepository
+import de.rki.coronawarnapp.server.protocols.internal.evreg.EventOuterClass
+import de.rki.coronawarnapp.server.protocols.internal.evreg.SignedEventOuterClass
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import org.joda.time.DateTime
+import org.joda.time.format.DateTimeFormat
+import timber.log.Timber
+import java.util.UUID
+
+class CreateEventTestViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    private val hostedEventRepository: HostedEventRepository
+) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
+
+    @AssistedFactory
+    interface Factory : SimpleCWAViewModelFactory<CreateEventTestViewModel>
+
+    val result = MutableLiveData<Result>()
+
+    fun createEvent(
+        description: String,
+        location: String,
+        start: String,
+        end: String,
+        defaultCheckInLengthInMinutes: String
+    ) {
+        try {
+            val startDate =
+                if (start.isBlank()) null else DateTime.parse(start, DateTimeFormat.forPattern("yyyy-MM-dd HH:mm"))
+            val endDate =
+                if (end.isBlank()) null else DateTime.parse(end, DateTimeFormat.forPattern("yyyy-MM-dd HH:mm"))
+
+            val startTimeStampMillis = startDate?.toInstant()?.millis ?: 0
+            val endTimeStampMillis = endDate?.toInstant()?.millis ?: 0
+
+            // Backend needs UNIX timestamp in Seconds, not milliseconds
+            val startTimeStampSeconds = (startTimeStampMillis / 1000).toInt()
+            val endTimeStampSeconds = (endTimeStampMillis / 1000).toInt()
```

Comment:
we hove an extension for that `instant.seconds`

Reply:
Nice, thank you for the tip. I adapted the code.